### PR TITLE
Feature/sleep-repository-and-log

### LINKED ISSUE #30: Calculate the average sleep quality for all users
**Description**
Calculate all time average sleep quality for all users

### LINKED ISSUE #31: Find all users who average a sleep quality greater than 3 for any given week (7 days)
**Description**
A method for `SleepRepository` class

### LINKED ISSUE #32: For a given day (identified by the date), find the users who slept the most number of hours
**Description**
Method for `SleepRepository`

### LINKED ISSUE #33: Instantiate sleep log for particular user
**Related Problem(s)**
Call method from within `SleepRepository` method because calculation without `SleepLog` method is far less efficient

**Description**
Method on `User` class

### LINKED ISSUE #34: For user, calculate the average number of hours slept per day

### LINKED ISSUE #36: For user, calculate average sleep quality per day over all time

### LINKED ISSUE #37: For user, calculate how many hours slept  + quality of sleep each day over the course of a given week (7 days)
**Description**
Return object with week of dates corresponding either to sleep quality or sleep length.

### LINKED ISSUE #38: For user, calculate how many hours and how well they slept for a specific day
**Description**
Method for `SleepLog` class. Should take date as argument.

### PR DESCRIPTION
**Pull request by kellydinneen**
#### Description
- `SleepRepository` class passes all tests: handles calculations that average all user sleep data or find users with specific sleep data characteristics among all user sleep data. Specifically, this class is responsible for:
     *  calculating the average sleep quality for all users #30
     * Finding all users who average a sleep quality greater than 3 for any given week (7 days) #31                
     * For a given day, finding the users who slept the greatest number of hours #32 
<br>

- `SleepLog` class is drafted in order to support `SleepRepository` methods, but has not yet been tested.

- PR also includes method in `User` class to instantiate `SleepLog` -- supports functioning of `findWeeksGoodSleepers()` method in `SleepRepository`

#### Functionality area
- styling
- html layout
- **data model**
- DOM
- data model-DOM interaction
- documentation

#### Where a reviewer should start
- SleepRepository.js 
- SleepLog.js 

#### How to test this change
- run `npm test test/SleepRepository-test.js`

#### Next Steps/Loose Ends
- need to test `SleepLog` class

### CODE REVIEW

**Comment by richardltyler** [comment]
- [x] Follows Turing Style Guidelines
- [x] Changes generate no new warnings or errors
- [x] No spelling errors
- [x] Approved changes with team
- [x] All teammates understand changes

File: src/SleepLog.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+class SleepLog {
+
+  constructor(sleepData, userID) {
+    this.log = sleepData.filter((data) => data.userID === userID)
+      .map(userLog => {
```
Comment (richardltyler) [inline src/SleepLog.js:5]: starting with this .map, I see why you're doing this. It makes sense to remove the ID from the object, considering we really aren't going to use it in this class. But I think this is extra code that doesn't really do anything productive.

File: src/User.js [hunk 2]
```diff
@@ -13,6 +15,10 @@ class User {
     const names = this.name.split(' ');
     return names[0];
   }
+
+  getSleepLog(sleepData) {
+    return new SleepLog(sleepData, this.id);
```
Comment (richardltyler) [inline src/User.js:20]: I think we should be storing this as a property like ```this.sleepLog = new SleepLog``` that way we don't have to call the function and reinstantiate the class each time we need to use the value

**Comment by kellydinneen** [comment]
Refactors pushed -- Ready to merge!

**Pull request merged**